src/xrpld/rpc/detail/RPCCall.cpp

```diff
@@ -266,24 +298,50 @@ class RPCParser
         }
         else
         {
-            std::int64_t uLedgerMin = jvParams[1u].asInt();
-            std::int64_t uLedgerMax = jvParams[2u].asInt();
+            std::int32_t ledgerMin, ledgerMax;
+            if (auto ledgerMinOpt = jvParseInt(jvParams[1u]))
+            {
+                ledgerMin = *ledgerMinOpt;
+            }
+            else
+            {
+                return rpcError(rpcINVALID_LGR_RANGE);
+            }
 
-            if (uLedgerMax != -1 && uLedgerMax < uLedgerMin)
+            if (auto ledgerMaxOpt = jvParseInt(jvParams[2u]))
```

Comment:
nit: This too can be const

```suggestion
            if (auto const ledgerMaxOpt = jvParseInt(jvParams[2u]))            
```

Reply:
Also shouldn't we parse this value as uint, thus avoiding the `-1` later on?

Reply:
No, `-1` is a valid value. https://xrpl.org/docs/references/http-websocket-apis/public-api-methods/account-methods/account_tx
```
A value of -1 instructs the server to use the most recent validated ledger version available.
```